src/assets/scripts/client/game/GameController.js

```diff
@@ -437,25 +446,59 @@ class GameController {
      * @return {number}
      */
     getPtlLength() {
-        const currentPtlVal = this.getGameOption(GAME_OPTION_NAMES.PTL_LENGTH);
+        let userSettingsPtlLength = this.getGameOption(GAME_OPTION_NAMES.PROJECTED_TRACK_LINE_LENGTH);
+
+        if (userSettingsPtlLength === 'from-theme') {
+            userSettingsPtlLength = this.theme.RADAR_TARGET.PROJECTED_TRACK_LINE_LENGTH;
+        }
 
-        return parseFloat(currentPtlVal);
+        return parseFloat(userSettingsPtlLength);
     }
-    
+
     /**
      * Check whether or not the trailing distance separator should be drawn.
      *
-     * Used by the `CanvasController` to determine whether or not to proceed with 
+     * Used by the `CanvasController` to determine whether or not to proceed with
      * `canvas_draw_separation_indicator`.
      *
      * @for GameController
-     * @method shouldUseTrailingSeparator
-     * @param aircraft
+     * @method shouldUseTrailingSeparationIndicator
+     * @param aircraft {AircraftModel}
      * @return {boolean}
      */
-    shouldUseTrailingSeparator(aircraft) {
-        return aircraft.isDeparture() && this.getGameOption(GAME_OPTION_NAMES.DRAW_ILS_DISTANCE_SEPARATOR) === 'yes';
+    shouldUseTrailingSeparationIndicator(aircraft) {
+        const isArrival = aircraft.isArrival();
+        const userSettingsValue = this.getGameOption(GAME_OPTION_NAMES.DRAW_ILS_DISTANCE_SEPARATOR);
+        let isIndicatorEnabled = userSettingsValue === 'yes';
+
+        if (userSettingsValue === 'from-theme') {
+            isIndicatorEnabled = this.theme.RADAR_TARGET.TRAILING_SEPARATION_INDICATOR_ENABLED;
+        }
+
+        return isArrival && isIndicatorEnabled;
     }
+
+    // TODO: Upon removal of `this.getPtlLength()`, this will no longer be needed
```

Comment:
not sure, but i'm too exhausted to rip the PTLs out of the game controller now, so sometime in the distant future I guess.